server/controllers/admin/roles.js

```diff
@@ -151,6 +141,71 @@ function listForUser(req, res, next) {
     .done();
 }
 
+
+function rolesAction(req, res, next) {
+
+  const roleUuid = db.bid(req.params.roleUuid);
+  const sql = `
+    SELECT a.id, a.description, IFNULL(s.affected, 0) as affected
+    FROM actions a
+    LEFT JOIN(
+      select  actions_id , 1 as affected 
+      FROM role_actions ra
+      JOIN role ro ON ra.role_uuid = ro.uuid
+      WHERE ro.uuid = ?
+    )s ON s.actions_id = a.id
+  `;
+
+  db.exec(sql, [roleUuid])
+    .then((actions) => {
+      res.json(actions);
+    })
+    .catch(next)
+    .done();
+}
+
+
+// affect roles to a user
+// roles ares permissions
+function assignActionToRole(req, res, next) {
+  const data = req.body;
+
+  const actionIds = [].concat(data.action_ids);
+
+  const roleUuid = db.bid(data.role_uuid);
+
+
+  const deleleUserRoles = `DELETE FROM role_actions WHERE role_uuid=? `;
+  const addAction = `INSERT INTO role_actions SET ?`;
+
+  db.exec(deleleUserRoles, roleUuid)
+    .then(() => {
+      const promisses = actionIds.map(actionId => {
+        return db.exec(addAction, { uuid : db.uuid(), role_uuid : roleUuid, actions_id : actionId });
+      });
+
+      return q.all(promisses);
+    }).then(() => {
+      res.sendStatus(201);
+    }).catch(next)
+    .done();
+}
+
+function hasAction(req, res, next) {
+  const actionId = req.params.action_id;
+  const userId = req.session.user.id;
+
+  const sql = `
+    SELECT count(ra.uuid) as nbr FROM role_actions ra
+    JOIN user_role as ur ON ur.role_uuid = ra.role_uuid
+    WHERE actions_id =? AND ur.user_id=?`;
+
+  db.one(sql, [actionId, userId]).then(row => {
+    res.json(row.nbr > 0);
```

Comment:
Always include a status code with your responses.  I'm surprised express doesn't throw a warning on this route...

Something like:
```js
res.status(200).send(row.nbr > 0);
```